controllers/podchaos/podfailure/types.go

```diff
@@ -96,22 +86,17 @@ func (r *Reconciler) Apply(ctx context.Context, req ctrl.Request, chaos twophase
 			HostIP:    pod.Status.HostIP,
 			PodIP:     pod.Status.PodIP,
 			Action:    string(podchaos.Spec.Action),
-			Message:   fmt.Sprintf(podFailureActionMsg, *podchaos.Spec.Duration),
 		}
-
+		if podchaos.Spec.Duration != nil {
+			ps.Message = fmt.Sprintf(podFailureActionMsg, *podchaos.Spec.Duration)
+		}
 		podchaos.Status.Experiment.Pods = append(podchaos.Status.Experiment.Pods, ps)
 	}
-
 	return nil
 }
 
-func (r *Reconciler) Recover(ctx context.Context, req ctrl.Request, chaos twophase.InnerObject) error {
-	podchaos, ok := chaos.(*v1alpha1.PodChaos)
-	if !ok {
-		err := errors.New("chaos is not PodChaos")
-		r.Log.Error(err, "chaos is not PodChaos", "chaos", chaos)
-		return err
-	}
+// Clean would recover the podchaos for the selected pods
+func (r *Reconciler) Clean(ctx context.Context, req ctrl.Request, podchaos *v1alpha1.PodChaos) error {
```

Comment:
I think it's is more suitable to put type assertion logic down to `Clean` and `Perform`. There are redundant codes in `duration/podfailure/types.go` and `twophase/podfailure/types.go`

Reply:
updated.